spark/src/main/java/org/opensearch/sql/spark/flint/operation/FlintIndexOp.java

```diff
@@ -145,11 +148,18 @@ public void cancelStreamingJob(
     String jobId = flintIndexStateModel.getJobId();
     try {
       emrServerlessClient.cancelJobRun(
-          flintIndexStateModel.getApplicationId(), flintIndexStateModel.getJobId());
-    } catch (IllegalArgumentException e) {
-      // handle job does not exist case.
+          flintIndexStateModel.getApplicationId(), flintIndexStateModel.getJobId(), true);
+    } catch (ResourceNotFoundException e) {
```

Comment:
This happens when the current JobId is not found?
Should we ignore this or throw exception?